src/Lib.P2P/Multiplex/Muxer.cs

```diff
@@ -0,0 +1,340 @@
+                                    };
+                                    await hdr.WriteAsync(Channel, cancel).ConfigureAwait(false);
+                                    Channel.WriteByte(0); // length
+                                    await Channel.FlushAsync(cancel).ConfigureAwait(false);
+                                }
+                            }
+#endif
+                            break;
+
+                        case PacketType.MessageInitiator:
+                            if (substream == null)
+                            {
+                                _log.Warn($"Message to unknown stream #{header.StreamId}");
+                                continue;
+                            }
+
+                            substream.AddData(payload);
+                            break;
+
+                        case PacketType.MessageReceiver:
+                            if (substream == null)
+                            {
+                                _log.Warn($"Message to unknown stream #{header.StreamId}");
+                                continue;
+                            }
+
+                            substream.AddData(payload);
+                            break;
+
+                        case PacketType.CloseInitiator:
+                        case PacketType.CloseReceiver:
+                        case PacketType.ResetInitiator:
+                        case PacketType.ResetReceiver:
+                            if (substream == null)
+                            {
+                                _log.Warn($"Reset of unknown stream #{header.StreamId}");
+                                continue;
+                            }
+
+                            substream.NoMoreData();
+                            Substreams.TryRemove(substream.Id, out _);
+                            SubstreamClosed?.Invoke(this, substream);
+                            break;
+
+                        default:
+                            throw new InvalidDataException($"Unknown Muxer packet type '{header.PacketType}'.");
+                    }
+                }
+            }
+            catch (EndOfStreamException)
+            {
+                // eat it
+            }
+            catch (IOException)
+            {
+                // eat it
+            }
+            catch (SocketException e) when (e.SocketErrorCode == SocketError.ConnectionReset)
+            {
+                // eat it
+            }
+            catch (Exception) when (cancel.IsCancellationRequested)
+            {
+                // eat it
+            }
+            catch (Exception e)
+            {
+                // Log error if the channel is not closed.
+                if (Channel.CanRead || Channel.CanWrite) _log.Error("failed", e);
+            }
+
+            // Some of the tests do not pass a connection.
+            if (Connection != null)
+                Connection.Dispose();
+            else if (Channel != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the missing 'else' clause.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)